src/setup.c

```diff
@@ -255,6 +266,55 @@ static void setup_encoders(void)
 	configure_timer_as_quadrature_encoder(TIM4);
 }
 
+/**
+ * @brief Setup SPI.
+ *
+ * SPI is configured as follows:
+ *
+ * - Master mode.
+ * - Clock baud rate: PCLK1 / speed_div; PCLK1 = 36MHz.
+ * - Clock polarity: 0 (idle low; leading edge is a rising edge).
+ * - Clock phase: 0 (out changes on the trailing edge and input data
+ *   captured on rising edge).
+ * - Data frame format: 8-bits.
+ * - Frame format: MSB first.
+ *
+ * NSS is configured to be managed by software.
+ */
+static void setup_spi(uint8_t speed_div)
+{
```

Comment:
On Bulebule the gpio_set_mode functions needed where on this function... move to gpio module also there?

Reply:
Oh, I get it. Yeah, we should probably move all GPIO-related functions to the `setup_gpio` function.

https://github.com/Bulebots/bulebule/issues/399